webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -575,6 +513,72 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     ],
   );
 
+  const handleActionSuccess = useCallback(
+    (action: ExperimentAction, successfulIds: number[]): void => {
+      const idSet = new Set(successfulIds);
+      const updateExperiment = (updated: Partial<ExperimentItem>) => {
+        setExperiments((prev) =>
+          prev.map((expLoadable) =>
+            Loadable.map(expLoadable, (experiment) =>
+              idSet.has(experiment.experiment.id)
+                ? { ...experiment, experiment: { ...experiment.experiment, ...updated } }
+                : experiment,
+            ),
+          ),
+        );
+      };
+      switch (action) {
+        case ExperimentAction.Activate:
+          updateExperiment({ state: RunState.Active });
+          break;
+        case ExperimentAction.Archive:
+          updateExperiment({ archived: true });
+          break;
+        case ExperimentAction.Cancel:
+          updateExperiment({ state: RunState.StoppingCanceled });
+          break;
+        case ExperimentAction.Kill:
+          updateExperiment({ state: RunState.StoppingKilled });
+          break;
+        case ExperimentAction.Pause:
+          updateExperiment({ state: RunState.Paused });
+          break;
+        case ExperimentAction.Unarchive:
+          updateExperiment({ archived: false });
+          break;
+        case ExperimentAction.Move:
+        case ExperimentAction.Delete:
+          setExperiments((prev) =>
+            prev.filter((expLoadable) =>
+              Loadable.match(expLoadable, {
+                Loaded: (experiment) => !idSet.has(experiment.experiment.id),
+                NotLoaded: () => true,
+              }),
+            ),
+          );
+          break;
+        // Exhaustive cases to ignore.
+        case ExperimentAction.CompareTrials:
+        case ExperimentAction.ContinueTrial:
+        case ExperimentAction.DownloadCode:
+        case ExperimentAction.Edit:
+        case ExperimentAction.Fork:
+        case ExperimentAction.HyperparameterSearch:
+        case ExperimentAction.OpenTensorBoard:
+        case ExperimentAction.SwitchPin:
+        case ExperimentAction.ViewLogs:
+          break;
+      }
+      handleSelectionChange('remove-all', [0, selectedExperimentIds.size]);
```

Comment:
this line `handleSelectionChange('remove-all', [0, selectedExperimentIds.size]);` is the only change in this function and move this func to resolve `Block-scoped variable 'handleSelectionChange' used before its declaration` error